packages/google_sign_in/google_sign_in_platform_interface/lib/src/utils.dart

```diff
@@ -0,0 +1,24 @@
+import '../google_sign_in_platform_interface.dart';
+
+GoogleSignInUserData nativeUserDataToPluginUserData(Map<String, dynamic> data) {
```

Comment:
nit: Consider making these static methods on `GoogleSignInUserData` and `GoogleSignInTokenData` called `from`. This seems to be the common pattern in plugins repos: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_core/lib/src/firebase_options.dart#L23

Reply:
I also have a slightly different implementation of these for the responses from the web plugin, do you suggest two different from methods in that case?

Reply:
This one:

https://github.com/ditman/plugins/pull/101/files#diff-7a3311403df2c2a963a7925eed770b07R28

Reply:
We discussed this offline and decided to leave them as separate methods instead of constructors, to be able to return null, and to make them something we can directly use from the promise, so:

```
return channel.invokeMapMethod<String, dynamic>('signIn')
    .then((Map<String, dynamic> data) => convertUserDataBlah(data));
```

ended up being:

```
return channel
    .invokeMapMethod<String, dynamic>('signIn')
    .then(getUserDataFromMap);
```

(As it should have been from the start, to be honest!)